core/Tracker/Action.php

```diff
@@ -228,6 +229,16 @@ protected function getUrlAndType()
         return false;
     }
 
+    public function setCustomField($field, $value)
```

Comment:
This was needed in order to add any fields for the tracker from a `RequestProcessor`. We could not use a `Dimension` here since Dimension work only for one column, but `CustomDimensions` plugin adds data for many columns under circumstances. We could create a `Dimension` classes dynamically in the plugin but this would be a bit fragile, at least for now.